vectordatasource/transform.py

```diff
@@ -3264,9 +3264,36 @@ def _merge_polygons(polygon_shapes):
     if not list_of_polys:
         return []
 
-    result = shapely.ops.unary_union(list_of_polys)
+    # first, try to merge the polygons as they are.
+    try:
+        result = shapely.ops.unary_union(list_of_polys)
+        return [result]
 
-    return [result]
+    except ValueError:
+        pass
+
+    # however, this can lead to numerical instability where polygons _almost_
+    # touch, so sometimes buffering them outwards a little bit can help.
+    try:
+        from shapely.geometry import JOIN_STYLE
+
+        # don't buffer by the full pixel, instead choose a smaller value that
+        # shouldn't be noticable.
+        buffer_size = tolerance / 16.0
```

Comment:
Would it make sense to pre-divide the tolerance value so that python doesn't have to do floating point divisions for every pass through here?

Reply:
That's a good idea, thanks.

This method is called pretty dynamically; the caller can configurably call either this, `_merge_polygons_with_buffer` or `_merge_lines` and they all do different things with `tolerance`. We could have a `tolerance` and `tolerance_16th` stored in the `RecursiveMerger` object, but that seems a little inelegant.

Also, I think the time saving from not repeating the division is probably dwarfed by the attempt (and failure, to get to this branch) to `unary_union` the list of polygons. For example, the `11/1052/672` tile that was failing, had over 4,000 polygons in that list!

What do you think?

Reply:
I was thinking this was called more frequently. You're right – let's keep it clean and as-is.